avocado/core/test.py

```diff
@@ -337,13 +429,74 @@ def datadir(self):
         host data files that are generic enough to be used for all tests
         contained in a given test file.
         """
+        # TODO: on removal of Test.datadir, move its code over to
+        # get_file_data_dir
         # Maximal allowed file name length is 255
         if (self.filename is not None and
                 len(os.path.basename(self.filename)) < 251):
             return self.filename + '.data'
         else:
             return None
 
+    def get_data(self, filename):
+        """
+        Returns the path of a data file, if one exists
+
+        This method respects the existence of directories in the
+        following order:
+
+        1) "$(TEST_FILENAME).data/$(TEST)/$(VARIANT-$HASH)", if a
```

Comment:
This should be `$(TEST_DATA_DIR)/$(TEST_NAME)/$(VARIANT_ID)`. Also I think it'd be easier to follow it without the brackets. For the `non-hash` variant we'd have to find the suitable name, but we'd have to do that anyway as that must be implemented as a generic concept (like `VARIANT_ID` and `VARIANT_NAME`).

Reply:
PS: I'd be fine with the `$TEST_FILENAME.data` if we defined it that way, which means disallowing to override the `data_dir` property... So basically we can only do that after we deprecate it. Still I'd probably prefer allowing people to specify custom base `data_dir` independent on the `filename`...

Reply:
Since we'll dropping the non-hash variant version for now, that's one less problem.  I'll use names that are more consistent with the names used elsewhere too.